File: from-1401-to-1500/1457-pseudo-palindromic-paths-in-a-binary-tree/solution.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
 func pseudoPalindromicPaths (root *TreeNode) int {
    result := 0
    var dfs func(node *TreeNode, seen map[int]bool)
    dfs  = func(node *TreeNode, seen map[int]bool){
        if seen[node.Val] {
            delete(seen, node.Val)
        } else {
            seen[node.Val] = true
        }

        seenCopy := map[int]bool{}
        for k, v := range seen {
            seenCopy[k] = v
        }
        if node.Left != nil {
            dfs(node.Left, seenCopy)
        }

        seenCopy = map[int]bool{}
        for k, v := range seen {
            seenCopy[k] = v
        }
        if node.Right != nil {
            dfs(node.Right, seenCopy)
        }

        if node.Left == nil && node.Right == nil {
            if len(seen) < 2 {
                result += 1
            }
        }
    }
    dfs(root, map[int]bool{})
    return result
}

File: from-1401-to-1500/1457-pseudo-palindromic-paths-in-a-binary-tree/solution.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def pseudoPalindromicPaths (self, root: Optional[TreeNode]) -> int:
        ans = 0
        def dfs(node, seen):
            nonlocal ans
            if node.val in seen:
                seen.remove(node.val)
            else:
                seen.add(node.val)
                
            if node.left:
                dfs(node.left, seen.copy())
            if node.right:
                dfs(node.right, seen.copy())
                
            if not node.left and not node.right:
                if len(seen) < 2:
                    ans += 1
                    
        dfs(root, set())            
        return ans
    